_tools/fetch_riot_data.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import json
+import re
+import shlex
+import subprocess
+import glob
+from string import Template
+
+try:
+    import requests
+except ImportError:
+    requests = None
+
+from default_data import (
+    DEFAULT_BOARDS, DEFAULT_CPUS, DEFAULT_DRIVERS,
+    DEFAULT_DRIVER_CATEGORIES, DEFAULT_CONTRIBUTORS, DEFAULT_STATS
+)
+
+DATA_DIR = os.getenv("DATA_DIR", os.path.join(os.path.abspath(__file__)))
+DEFAULT_RIOT_BASE = os.path.join(os.path.abspath(__file__), "_RIOT")
+RIOTBASE = os.path.abspath(os.getenv("RIOTBASE", DEFAULT_RIOT_BASE))
+
+RIOT_DOC_BASE_URL = "https://doc.riot-os.org"
+TEMPLATE_BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+#
+# Data files templates
+#
+BOARDS_TEMPLATE = """- name: ${name}
+  apiurl: ${doc_url}/group__boards__${group}.html
+"""
+
+CPU_TEMPLATE = """- name: ${name}
+  apiurl: ${doc_url}/group__cpu__${group}.html
+"""
+
+DRIVERS_TEMPLATE = """- name: ${name}
+  parent: drivers_${parent}
+  apiurl: ${doc_url}/group__drivers_${group}.html
+"""
+
+DRIVER_CATEGORIES_TEMPLATE = """- name: ${name}
+  group: drivers_${group}
+  apiurl: ${doc_url}/group__drivers_${group}.html
+"""
+
+CONTRIBUTORS_TEMPLATE = """- login: ${login}
+  avatar_url: ${avatar_url}
+  html_url: https://github.com/${login}
+"""
+
+STATS_TEMPLATE = """boards: ${boards}
+cpus: ${cpus}
+commits: ${commits}
+"""
+
+TEMPLATES = {
+    "boards": BOARDS_TEMPLATE,
+    "cpus": CPU_TEMPLATE,
+    "drivers": DRIVERS_TEMPLATE,
+    "drivers_cats": DRIVER_CATEGORIES_TEMPLATE,
+    "contributors": CONTRIBUTORS_TEMPLATE,
+    "stats": STATS_TEMPLATE,
+}
+
+
+CONTRIBUTORS_URL = "https://api.github.com/repos/RIOT-OS/RIOT/contributors"
+OBJ_PER_PAGE = 100
+MAX_PAGES = 20
+
+
+def fetch_contributors_data():
+    """Fetch the complete list of RIOT contributors via the GitHub API."""
+    # Return default contributors if requests module is not available
+    if requests is None:
+        return DEFAULT_CONTRIBUTORS
```

Comment:
Should we inform the user that dummy data is being returned, and the missing requirements should be installed to fix this?

Reply:
Maybe the `install_python_dependencies` target can be documented in the README

Reply:
Added the message and a note in the README.